lib/snitcher.rb

```diff
@@ -86,12 +86,18 @@ def snitch(*args)
   def initialize_opts(options, uri)
     timeout = options.fetch(:timeout, 5)
 
-    {
-      open_timeout: timeout,
-      read_timeout: timeout,
-      ssl_timeout:  timeout,
-      use_ssl:      use_ssl?(uri),
-    }
+    [
+      options.fetch(:p_addr, :ENV),
```

Comment:
My PR starts to have sense for ruby < 2.5

Maybe I can detect the ruby version here and sets
- nil for Ruby < 2.5
- :ENV for Ruby > 2.5